File: 08_May_2nd/Seonghun/PascalTriangle.java
public class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        List<List<Integer>> rowsList = new ArrayList<List<Integer>>();
        
        for (int row = 0 ; row < numRows ; row++) {
            
            List<Integer> rowList = new ArrayList<Integer>(row+1);
            
            for (int col = 0 ; col <= row ; col++) {
                
                if (col == 0 || col == row) 
                    rowList.add(1);
                else
                    rowList.add(rowsList.get(row-1).get(col-1) + rowsList.get(row-1).get(col));
            }
            
            rowsList.add(rowList);
        }
        
        return rowsList;
    }
}

File: 08_May_2nd/gyeongwook/pascalTriangle.py
class Solution:
    # @param {integer} numRows
    # @return {integer[][]}
    def generate(self, numRows):
        if numRows == 0 : return []
        
        triangle = []
        triangle.append([1])
        
        for length in range(2, numRows+1) :
            current_row = []
            
            for idx in range(length) :
                left = triangle[length-2][idx-1] if idx != 0 else 0
                right = triangle[length-2][idx] if idx != length-1 else 0
                current_row.append(left + right)  
                
            triangle.append(current_row)
            
        return triangle